dagstore_async.go

```diff
@@ -134,5 +136,32 @@ func (d *DAGStore) initializeShard(ctx context.Context, s *Shard, mnt mount.Moun
 		return
 	}
 
+	// add all cids in the shard to the inverted (cid -> []Shard Keys) index.
+	iterableIdx, ok := idx.(index.IterableIndex)
+	if !ok {
+		_ = d.failShard(s, d.completionCh, "shard index is not iterable")
+		return
+	}
+
+	mhIter := &mhIdx{iterableIdx: iterableIdx}
+	if err := d.invertedIndex.AddMultihashesForShard(mhIter, s.key); err != nil {
+		_ = d.failShard(s, d.completionCh, "failed to add shard multihashes to the inverted index: %w", err)
+		return
```

Comment:
Same as above. Failing here makes the shard totally inaccessible, even if the caller has the shard ID.

Reply:
Yes, but a database should be able to update all it's indices correctly before declaring an insert to be a success. We don't want to end up in a partial success case here where some indices have the latest and some do not.

Reply:
That assumes that all accesses require querying the top-level index, which isn't the case.